oximeter/oximeter/src/types.rs

```diff
@@ -591,6 +598,22 @@ impl Sample {
     pub fn metric_fields(&self) -> &Vec<Field> {
         &self.metric.fields
     }
+
+    // Check validity of field names for the target and metric. Currently this
+    // just verifies there are no duplicate names between them.
+    fn verify_field_names(
+        target: &FieldSet,
+        metric: &FieldSet,
+    ) -> Result<(), MetricsError> {
+        for name in target.fields.iter().map(|f| f.name.as_str()) {
+            if metric.fields.iter().any(|f| f.name == name) {
```

Comment:
WDYT about making the `fields` member of `FieldSet` a BTreeSet?

This operation seems like it could be O(N), but I think it's currently  O(N**2)? I dunno if N is large enough yet to make a significant difference, but if we're planning on doing lookups regularly, it might be worthwhile.